oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+  underlying tooling, this has become more laborious to keep running. Especially because
+  there are few people with Admin rights to Transifex and knowledge of the Transifex API;
+  this could become a recurring problem with each Open edX release.
+
+.. _ecommerce-scripts: https://github.com/openedx/ecommerce-scripts/tree/master/transifex
+.. _this pull request: https://github.com/openedx/edx-platform/pull/30567
+
+Rationale for migrating to the `Transifex GitHub App`_
+******************************************************
+
+* This is an upgrade of a system we use regularly, but do not want to have to maintain
+  regularly.
+* Upgrading from a bot (machine user) to an app/workflow is recommended by GitHub and
+  makes the translation process more open source. 
+* The `Transifex GitHub App`_ is developed and supported by Transifex
+* The `Transifex GitHub App`_ is very simple to configure and has many options. We can
+  set Transifex Projects to automatically upload/download translation files from a
+  repository once the translations are reviewed and accepted.
+* By using an app that is maintained by Transifex the organization, we reduce the
+  maintenance burden and are more future proof of changes they might make since they
+  maintain both the API and the `Transifex GitHub App`_.
+
+Rationale for consolidating translations files centrally
+********************************************************
+
+* Transifex only allows a one-to-one relationship between repositories and Transifex
+  Projects. Organizing all of the translation files into one repository and one Transifex
+  Project has a lower labor cost: projects are managed separately so we end up spending
+  less time tracking translation progress, and debugging translation issues when all
+  translation files are put in the same place. By decreasing the number of projects we
+  need to maintain, we can add more content like the MFE translations.
+* A repository that only contains text/binary files, and uses branches to separate
+  translations related to Open edX releases can make all interactions with translations
+  very quick and simple due to the ability to clone and sparse-checkout the branch of a
+  specific release and the directory (repository name) with translation files.
+
+Proposed Implementation
+***********************
+
+Move Translation Files to a New Repo
+====================================
+
+Translation files (of types .mo, .po, and .json) currently exist amongst the code and
+documentation they translate. We will move these translation files from being amongst the
+code/documentation to their own repo. For example, a translation file for the openedx
+repository `edx-platform`_ located at
+``edx-platform/conf/locale/en/LC_MESSAGES/django.po`` would be moved to the new
+repository with the name openedx-translations_ located at
+``openedx-translation/edx-platform/conf/locale/en/LC_MESSAGES/django.po``. For easier
```

Comment:
Can I suggest to create an intermediate "repos" folder, such that we have `openedx-translations/repos/edx-platform`? Otherwise we will have a lot of repos located at the root, as we already do: https://github.com/openedx/openedx-translations This will quickly become unwieldy. And if we need extra translations-specific folders at the root, they will be mixed with repo names.

Reply:
This is a good idea that we can implement once the review period is complete.